src/main/kotlin/com/ampnet/identityservice/service/impl/AutoInvestQueueServiceImpl.kt

```diff
@@ -34,9 +36,12 @@ class AutoInvestQueueServiceImpl(
 
     companion object : KLogging() {
         const val QUEUE_NAME = "AutoInvestQueue"
+        const val SUPPORTED_VERSION = "1.0.20"
```

Comment:
I would extract all of the version comparison logic into a separate data class like this:
```kotlin
data class Version(val version: String) : Comparable<Version> {

    private val parts = version.split('.').mapNotNull { it.toIntOrNull() }

    fun requireSize(size: Int): Version {
        if (parts.size != size) {
            throw InvalidRequestException(ErrorCode.BLOCKCHAIN_UNSUPPORTED_VERSION, "Invalid contract version")
        }
        return this
    }

    override fun compareTo(other: Version): Int {
        tailrec fun compareVersion(index: Int, maxIndex: Int, other: Version): Int {
            val thisPart = parts.elementAtOrElse(index) { 0 }
            val thatPart = other.parts.elementAtOrElse(index) { 0 }
            val result = thisPart.compareTo(thatPart)

            return if (result != 0 || index >= maxIndex) {
                result
            } else {
                compareVersion(index + 1, maxIndex, other)
            }
        }

        val maxIndex = max(parts.size, other.parts.size)
        return compareVersion(0, maxIndex, other)
    }
}
```
You can then use `requireSize` to validate number of version parts. A nice practice for more purely functional code is to write loops using the tail recursion (which gets nicely optimized by the compiler).

Demonstration via main function:

```kotlin
fun main(args: Array<String>) {
    println(Version("1") > Version("2"))
    println(Version("1.0") > Version("2"))
    println(Version("1.0") > Version("1.0.1"))
    println(Version("1.0") > Version("1.0.0"))
    println(Version("1.2") > Version("1.0.1"))
    println(Version("1.0.1") > Version("1.0.0"))
}
```